idea/src/org/jetbrains/kotlin/idea/inspections/ForEachParameterNotUsed.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.descriptors.VariableDescriptor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.intentions.getCallableDescriptor
+import org.jetbrains.kotlin.idea.refactoring.getThisLabelName
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtVisitorVoid
+import org.jetbrains.kotlin.psi.callExpressionVisitor
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+
+class ForEachParameterNotUsed : AbstractKotlinInspection() {
+    companion object {
+        private val COLLECTIONS_FOREACH_FQNAME = FqName("kotlin.collections.forEach")
+        private val SEQUENCES_FOREACH_FQNAME = FqName("kotlin.sequences.forEach")
+    }
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return callExpressionVisitor {
+            when(it.getCallableDescriptor()?.fqNameOrNull()) {
```

Comment:
For performance reasons, it's a bad idea to get descriptor at the very first line. Here I recommend to check name first via PSI, and only after it check name via descriptors.

Reply:
Also, space is required after `when` keyword (code style requirements)